pkg/validation/internal/operatorhub.go

```diff
@@ -74,111 +96,303 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
-		errs = append(errs, fmt.Errorf("csv.Spec.Provider.Name not specified"))
 	}
 
-	for _, maintainer := range csv.Spec.Maintainers {
-		if maintainer.Name == "" || maintainer.Email == "" {
-			errs = append(errs, fmt.Errorf("csv.Spec.Maintainers elements should contain both name and email"))
-		}
-		if maintainer.Email != "" {
-			_, err := mail.ParseAddress(maintainer.Email)
-			if err != nil {
-				errs = append(errs, fmt.Errorf("csv.Spec.Maintainers email %s is invalid: %v", maintainer.Email, err))
-			}
+	if len(channelsNotFollowingConventional) > 0 {
+		return fmt.Errorf("channel(s) %+q are not following the recommended naming convention: "+
+			"https://olm.operatorframework.io/docs/best-practices/channel-naming",
+			channelsNotFollowingConventional)
+	}
+
+	return nil
+}
+
+// validateHubDeprecatedAPIS will check if the operator bundle is using a deprecated or no longer supported k8s api
+// Note if the k8s was informed via "k8s=1.22" it will be used. Otherwise, we will use the minKubeVersion in
+// the CSV to do the checks. So, the criteria is >=minKubeVersion. By last, if the minKubeVersion is not provided
+// then, we should consider the operator bundle is intend to work well in any Kubernetes version.
+// Then, it means that:
+//--optional-values="k8s=value" flag with a value => 1.16 <= 1.22 the validator will return result as warning.
+//--optional-values="k8s=value" flag with a value => 1.22 the validator will return result as error.
+//minKubeVersion >= 1.22 return the error result.
+//minKubeVersion empty returns a warning since it would mean the same of allow install in any supported version
+func validateHubDeprecatedAPIS(bundle *manifests.Bundle, versionProvided string) (errs, warns []error) {
+	// K8s version where the apis v1betav1 is no longer supported
+	const k8sVerV1betav1Unsupported = "1.22.0"
+	// K8s version where the apis v1betav1 was deprecated
+	const k8sVerV1betav1Deprecated = "1.16.0"
+	// semver of the K8s version where the apis v1betav1 is no longer supported to allow us compare
+	semVerK8sVerV1betav1Unsupported := semver.MustParse(k8sVerV1betav1Unsupported)
+	// semver of the K8s version where the apis v1betav1 is deprecated to allow us compare
+	semVerk8sVerV1betav1Deprecated := semver.MustParse(k8sVerV1betav1Deprecated)
+	// isVersionProvided defines if the k8s version to test against was or not informed
+	isVersionProvided := len(versionProvided) > 0
+
+	// Transform the key/option versionProvided in semver Version to compare
+	var semVerVersionProvided semver.Version
+	if isVersionProvided {
+		var err error
+		semVerVersionProvided, err = semver.ParseTolerant(versionProvided)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("invalid value informed via the k8s key option : %s", versionProvided))
+		} else {
+			// we might want to return it as info instead of warning in the future.
+			warns = append(warns, fmt.Errorf("checking APIs against Kubernetes version : %s", versionProvided))
```

Comment:
I still have a negative opinion about warning here. However I won't let it block the PR. This can always be changed later.